packages/ethersjs-provider/src/EthersProvider.ts

```diff
@@ -1,62 +1,99 @@
-            throw new Error(`Contract name ${contractName} not found in EthersJSProvider. Call 'storeAbiData' first with this name and the contract abi`);
-        }
-        return contractInterface.events[eventName].topic;
-    }
+  public async getBlockNumber(): Promise<number> {
+    return await super.getBlockNumber();
+  }
 
-    public parseLogValues(contractName: string, log: Log): LogValues {
-        const contractInterface = this.contractMapping[contractName];
-        if (!contractInterface) {
-            throw new Error(`Contract name ${contractName} not found in EthersJSProvider. Call 'storeAbiData' first with this name and the contract abi`);
-        }
-        const parsedLog = contractInterface.parseLog(log);
-        let omittedValues = _.map(_.range(parsedLog.values.length), (n) => n.toString());
-        omittedValues.push('length');
-        let logValues = _.omit(parsedLog.values, omittedValues);
-        logValues = _.mapValues(logValues, (val) => {
-            if (val._hex) {
-            return val._hex;
-            }
-            return val;
-        });
-        return logValues;
+  public storeAbiData(abi: Abi, contractName: string): void {
+    this.contractMapping[contractName] = new ethers.utils.Interface(abi);
+  }
+
+  public getEventTopic(contractName: string, eventName: string): string {
+    const contractInterface = this.contractMapping[contractName];
+    if (!contractInterface) {
+      throw new Error(`Contract name ${contractName} not found in EthersJSProvider. Call 'storeAbiData' first with this name and the contract abi`);
     }
+    return contractInterface.events[eventName].topic;
+  }
 
-    public async getLogs(filter: Filter): Promise<Array<Log>> {
-        return super.getLogs(filter);
+  public parseLogValues(contractName: string, log: Log): LogValues {
+    const contractInterface = this.contractMapping[contractName];
+    if (!contractInterface) {
+      throw new Error(`Contract name ${contractName} not found in EthersJSProvider. Call 'storeAbiData' first with this name and the contract abi`);
     }
+    const parsedLog = contractInterface.parseLog(log);
+    let omittedValues = _.map(_.range(parsedLog.values.length), (n) => n.toString());
+    omittedValues.push('length');
+    let logValues = _.omit(parsedLog.values, omittedValues);
+    logValues = _.mapValues(logValues, (val) => {
+      if (val._hex) {
+        return val._hex;
+      }
+      return val;
+    });
+    return logValues;
+  }
+
+  public async getLogs(filter: Filter): Promise<Array<Log>> {
+    return super.getLogs(filter);
+  }
+
+  public async perform(message: any, params: any): Promise<any> {
+    const _this = this;
+    return new Promise((resolve, reject) => {
+      retry(
+        this.opts,
+        async function (callback) {
+          let results: any;
+          try {
+            results = await _this.provider.perform(message, params);
+          } catch (err) {
+            return callback(err);
```

Comment:
After merging Justin's reformatting PR into this branch, I re-ran the tests and apparently adding `undefined` here appeared to work because this hadn't been rebuilt when I was running the test previously. The real fix for this was returning callback(err) instead of just calling it.